### PR TITLE
Verify HTML reports with puppeteer

### PR DESCRIPTION
**Pull request by aslakhellesoy**
This automates the verification of generated HTML reports by rendering them with puppeteer. If the rendering fails, the build fails.

In order to make it pass I also had to downgrade to webpack 4 - not sure what's up with webpack 5.

### CODE REVIEW

File: html-formatter/javascript/check.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const fs = require('fs')
+const path = process.argv[2]
+const html = fs.readFileSync(path, 'utf-8')
+const puppeteer = require('puppeteer');
+
+(async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  await page.setContent(html)
+  await page.pdf({path: 'hn.pdf', format: 'A4'});
```
Comment (davidjgoss) [inline html-formatter/javascript/check.js:10]: What purpose does the PDF generation serve?

File: html-formatter/javascript/check.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const dynamicHTML = await page.evaluate(() => {
+    const content = document.querySelector('.cucumber-react')
```
Comment (davidjgoss) [inline html-formatter/javascript/check.js:12]: Not sure if there could be errors in lower components (e.g. markdown) but the root one still render enough for this to pass?

To solve a similar problem before I've queried the console for errors - I could always do a follow up if you think it's a good idea.

**Review by davidjgoss — COMMENTED** [review]
Nice approach.

> In order to make it pass I also had to downgrade to webpack 4 - not sure what's up with webpack 5.

Yeah more detail on that in #1279, seems like Webpack's module resolution must have changed and not be able to handle these circular imports any more. I can't see a reference to it in [their announcement](https://webpack.js.org/blog/2020-10-10-webpack-5-release/) unless I'm not recognising the language.

(I've got a change in flight to address the circular imports in our code and add an eslint rule, so I'll leave that issue open.)

File: html-formatter/javascript/check.js (hunk 1 above)
Comment (aslakhellesoy) [reply html-formatter/javascript/check.js]: re: pdf. Copy paste error - removed

File: html-formatter/javascript/check.js (hunk 2 above)
Comment (aslakhellesoy) [reply html-formatter/javascript/check.js:11]: Yes, inspecting console errors could be useful too. This check does not guarantee absence of errors, but it does detect presence of errors. I think that's a good start.

But we have a problem with CircleCI launching puppeteer.....

Comment (davidjgoss) [reply html-formatter/javascript/check.js:11]: I've seen this before. They started requiring some libraries on the host system from 3.0.0 inc. `libgbm`.

**Review by davidjgoss — APPROVED** [review]

**Pull request merged**